File: exercise/day14/src/test/java/games/FizzBuzzTests.java
package games;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

class FizzBuzzTests {

    public static Stream<Arguments> validInputs() {
        return Stream.of(
                Arguments.of(1, "1"),
                Arguments.of(67, "67"),
                Arguments.of(82, "82"),
                Arguments.of(3, "Fizz"),
                Arguments.of(66, "Fizz"),
                Arguments.of(99, "Fizz"),
                Arguments.of(5, "Buzz"),
                Arguments.of(50, "Buzz"),
                Arguments.of(85, "Buzz"),
                Arguments.of(15, "FizzBuzz"),
                Arguments.of(30, "FizzBuzz"),
                Arguments.of(45, "FizzBuzz")
        );
    }

    public static Stream<Arguments> invalidInputs() {
        return Stream.of(
                Arguments.of(0),
                Arguments.of(-1),
                Arguments.of(101)
        );
    }

    @ParameterizedTest
    @MethodSource("validInputs")
    void returns_number_representation(int input, String expectedResult) throws OutOfRangeException {
        assertThat(FizzBuzz.convert(input))
                .isEqualTo(expectedResult);
    }

    @ParameterizedTest
    @MethodSource("invalidInputs")
    void throws_an_exception_for_numbers_out_of_range(int input) {
        assertThatThrownBy(() -> FizzBuzz.convert(input))
                .isInstanceOf(OutOfRangeException.class);
    }
}

File: exercise/c#/Day10/Day10.Tests/FizzBuzzTests.cs
using FluentAssertions;
using Xunit;

namespace Day10.Tests
{
    public class FizzBuzzTests
    {
        [Theory]
        [InlineData(1, "1")]
        [InlineData(67, "67")]
        [InlineData(82, "82")]
        [InlineData(3, "Fizz")]
        [InlineData(66, "Fizz")]
        [InlineData(99, "Fizz")]
        [InlineData(5, "Buzz")]
        [InlineData(50, "Buzz")]
        [InlineData(85, "Buzz")]
        [InlineData(15, "FizzBuzz")]
        [InlineData(30, "FizzBuzz")]
        [InlineData(45, "FizzBuzz")]
        public void Returns_Number_Representation(int input, string expectedResult)
            => FizzBuzz.Convert(input)
                .Should()
                .Be(expectedResult);

        [Theory]
        [InlineData(0)]
        [InlineData(101)]
        [InlineData(-1)]
        public void Fails_For_Numbers_Out_Of_Range(int input)
            => ((Action) (() => FizzBuzz.Convert(input)))
                .Should()
                .Throw<OutOfRangeException>();
    }
}